### PR TITLE
added more golang paths to search for `hexokinase` binary

### PR DESCRIPTION
**Pull request by BourgeoisBear**

### CODE REVIEW

File: plugin/hexokinase.vim [hunk 1]
```diff
@@ -8,14 +8,29 @@ endif
 
 let g:loaded_hexokinase = 1
 
-let g:Hexokinase_executable_path = get(g:, 'Hexokinase_executable_path',
-            \ executable(expand('<sfile>:h:h').'/hexokinase/hexokinase')
-            \   ? expand('<sfile>:h:h').'/hexokinase/hexokinase'
-            \   : expand($GOPATH).'/bin/hexokinase')
+function! hexokinase#get_exec_path()
+
+	let lPaths = [
```
Comment (RRethy) [inline plugin/hexokinase.vim]: snake-case for variables, also `lPaths` isn't a great variable name.

**Review by RRethy — CHANGES_REQUESTED** [review]

**Comment by BourgeoisBear** [comment]
how is this? (lambda var requires uppercase first letter BTW).

**Comment by BourgeoisBear** [comment]
After thinking about this, I've really just put another hack on top of your hack.  Using $PATH first would be the right thing to do.  Golang people have already appended their `go/bin` (wherever that may be) to the $PATH.  Non-golang people will have to install the compiler anyway, and we can always fall-back to the default `$HOME/go/bin` for them.

vim-go has a `:GoInstallBinaries` command to `go get` everything, honoring the user's preferred `go/bin` location.  As-is, getting this stowed to a new box is very manual.

Before I change anything, Yea or Nay?

**Comment by RRethy** [comment]
I'd rather not add much logic here, the dependency is vendor'd so there's no expectation that the user has it on their path. `$GOPATH` was just a relic from when the binary wasn't vendor'd. Right now, I'm actually no longer inclined to to merging this PR after giving it a bit more thought, is there a specific reason the vendor'd dependency doesn't work for you?

**Comment by BourgeoisBear** [comment]
Yes.  That's what I'm saying.  Abandon this pull request.  I'm suggesting a different approach.

1) Current user experience on new box:
    - open vim
    - 'make hexokinase' error
    - leave vim
    - cd .vim/pack/plugins/start/vim-hexokinase
    - make hexokinase
    - go back to vim & continue

2) Proposed experience:
    - open vim
    - 'please run :HexokinaseInstall' error 
    - :HexokinaseInstall (which just does a `go get github.com/RRethy/hexokinase` behind the scenes)

As for paths, with option #2, you can just lean on the system path and a proper golang installation `if executable('hexokinase')`, and if that fails, assume the default go build destination of `$HOME/go/bin`.

This is just a suggestion now, since I finished hot-rodding @lilydjwg's colorizer (https://github.com/BourgeoisBear/colorizer) to be fast enough, and it's pure vimscript, which solves my problem.

**Comment by RRethy** [comment]
For me those two experiences are the same, and this will be the case for anyone with a decent plugin manager which will handle this stuff for you. The issue with `go get` is a lack of versioning.

**Comment by BourgeoisBear** [comment]
pkg.go.dev

```sh
export GO111MODULE=on
go install github.com/RRethy/hexokinase@v0.0.0-20200121161234-b345a18ba387
```

**Pull request closed without merging**